data/model/oci/tag.py

```diff
@@ -801,6 +802,13 @@ def fetch_paginated_autoprune_repo_tags_by_number(
             .offset(tags_offset)
             .limit(items_per_page)
         )
+        if tag_pattern is not None:
+            query = db_regex_search(
+                Tag.select(query.c.name).from_(query),
+                query.c.name,
```

Comment:
what does `query.c.name` return? why are we querying for `Tag.select(query.c.name).from_(query)` here while we are just passing the query [here](https://github.com/quay/quay/pull/3188/files#diff-d54089ecfb2c9ed6312fb03ca1ea1cd1be9555c8dedf9138a2feac65ea530ff5R845)

Reply:
`query.c.name` is the same as `Tag.name` from the original query. It's being used as a subquery via the `.from_(query)` because we only want to match on tags that **exceed** the limit. Any tags that match the regex will still be counted towards the tag count limit.

Reply:
I am just trying to understand why you chose to use `query.c.name` here but `Tag.name` [here](https://github.com/quay/quay/pull/3188/files#diff-d54089ecfb2c9ed6312fb03ca1ea1cd1be9555c8dedf9138a2feac65ea530ff5R845)

Reply:
It's because of how peewee alias' the tables. The subquery is aliased as `T2` but the query references `Tag.name` as `T1.name` but since we're using a subquery in the `from` there is no `T1`. So that results in an error similar to `T1 does not exist`.

Reply:
ok, that makes sense.